addons/knowledge/static/src/components/permission_panel/permission_panel.js

```diff
@@ -181,147 +248,135 @@ class PermissionPanel extends Component {
-            const userId = userIds && userIds.length === 1 ? userIds[0] : false;
-
-            if (userId) {
-                const messaging = await this.env.services.messaging.get();
-                messaging.openChat({
-                    userId: userId
-                });
+    _openArticle (id) {
+        this.env.bus.trigger('do-action', {
+            action: 'knowledge.ir_actions_server_knowledge_home_page',
+            options: {
+                additional_context: {
+                    res_id: id
+                }
             }
-        }
+        });
     }
 
-  /**
-    * This method is called before each permission change rpc when the user needs to confirm the change as them
-    * would lose them access to the article if them do confirm.
-    * @param {str} message
-    * @param {function} confirm
-    * @param {function} discard
-    */
-    _showConfirmDialog (message, confirm, discard) {
-        if (discard === undefined) {
-            discard = this.loadPanel;
-        }
-        Dialog.confirm(this, message, {
+    /**
+     * Opens a confirmation dialog
+     * @param {Object} options
+     * @param {String} options.message
+     * @param {Function} options.confirm
+     * @param {Function} options.discard
+     */
+    _openConfirmationDialog (options) {
+        const { _t } = this.env;
+        return Dialog.confirm(this, options.message, {
             buttons: [{
-                text: _t('confirm'),
+                text: _t('Confirm'),
                 classes: 'btn-primary',
                 close: true,
-                click: confirm
+                click: options.confirm
             }, {
                 text: _t('Discard'),
                 close: true,
-                click: discard
+                click: options.discard
             }],
         });
     }
 
-  /**
-    * This method is called after each permission change rpc.
-    * It will check if a reloading of the article tree or a complete reload is needed in function
-    * of the new article state (if change of category or if user lost their own access to the current article).
-    * return True if the caller should continue after executing this method, and False, if caller should stop.
-    * @param {Dict} result
-    * @param {Boolean} lostAccess
-    */
-    _onChangedPermission (result, reloadAll, reloadArticleId) {
-        if (result.error) {
-            Dialog.alert(this, result.error,{
-              title: _t("Error"),
-            });
-        } else if (reloadAll && reloadArticleId) {  // Lose write access
-            this.openArticle(reloadArticleId);
-            return false;
-        } else if (reloadAll) {  // Lose access -> Hard Reload
-            window.location.replace('/knowledge/home');
-        } else if (result.reload_tree) {
-            this.env.bus.trigger('reload_tree', {});
-        }
-        return true;
+    /**
+     * Redirects the user to the homepage.
+     */
+    _redirectToHompage () {
```

Comment:
rename it to "_redirectToKnowledgeHome(page)' to avoid confusion with the Odoo App switcher